Valhalla_build_tiles command terminated by signal 11

Yesterday I tried to run this line

`valhalla_build_tiles -c valhalla.json north-america-latest.osm.pbf  europe-latest.osm.pbf`

It ran for about 13 hours before it terminated by signal 11. Is this a known bug? Should I try just running one at a time? I have successfully built North America tiles before, but this was the first time I've tried Europe. I, unfortunately, don't have the last few lines it output to the screen because I rebooted my server. Has anyone else seen this and hopefully, know how I can fix it?

Edit: I'm using Valhalla version 3.0.7

It failed again. Same error. The output looks like this:
2019/09/19 19:50:40.962851 [INFO] Start stage = initialize End stage = cleanup
2019/09/19 19:50:42.189504 [INFO] Parsing files: north-america-latest.osm.pbf, europe-latest.osm.pbf
2019/09/19 19:50:42.189561 [INFO] Parsing ways...
2019/09/19 19:53:55.427264 [WARN] Max osmid exceeded bitset, resizing to fit id: 6570550716
2019/09/19 19:53:56.238566 [WARN] Max osmid exceeded bitset, resizing to fit id: 6733378894
2019/09/19 19:53:56.251311 [WARN] Max osmid exceeded bitset, resizing to fit id: 6584833965
2019/09/19 19:53:56.764463 [WARN] Max osmid exceeded bitset, resizing to fit id: 6726943266
2019/09/19 19:53:56.905024 [WARN] Max osmid exceeded bitset, resizing to fit id: 6804527419
2019/09/19 19:53:58.322573 [WARN] Max osmid exceeded bitset, resizing to fit id: 6802243904
2019/09/19 20:42:27.479472 [INFO] invalid_argument thrown for condition (Jun 15-Labor Day)
2019/09/19 20:45:19.247502 [INFO] invalid_argument thrown for condition (Jun 15-Labor Day)
2019/09/19 22:44:22.666506 [INFO] invalid_argument thrown for condition (Jan 01- May 31)
2019/09/19 22:44:22.667555 [INFO] invalid_argument thrown for condition (Jan 01- May 31)
2019/09/19 23:40:03.695527 [INFO] invalid_argument thrown for condition (Jan 01- May 31)
2019/09/20 00:25:20.407829 [INFO] Finished with 88364744 routable ways containing 923122185 nodes
2019/09/20 00:25:20.407887 [INFO] Sorting osm access tags by way id...
2019/09/20 00:25:21.363803 [INFO] Parsing relations...
2019/09/20 00:37:03.232264 [INFO] invalid_argument thrown for condition  Jul 1- Aug 31 off
2019/09/20 01:12:51.751187 [INFO] Finished with 760483 simple restrictions
2019/09/20 01:12:51.751232 [INFO] Finished with 0 lane connections
2019/09/20 01:12:51.807951 [INFO] Sorting complex restrictions by from id...
2019/09/20 01:12:51.853557 [INFO] Sorting complex restrictions by to id...
2019/09/20 01:12:51.898987 [INFO] Sorting osm way node references by node id...
2019/09/20 01:33:31.962573 [INFO] Finished
2019/09/20 01:33:31.965992 [INFO] Parsing nodes...
2019/09/20 04:33:16.278260 [INFO] Finished with 793268492 nodes contained in routable ways
2019/09/20 04:33:16.278969 [INFO] Sorting osm way node references by way index and node shape index...
2019/09/20 04:55:28.951094 [INFO] Finished: max_osm_id 6804068107
2019/09/20 04:55:28.952547 [INFO] Number of nodes with refs (exits) = 81602
2019/09/20 04:55:28.952553 [INFO] Number of nodes with exit_to = 9535
2019/09/20 04:55:28.952557 [INFO] Number of nodes with names = 36327
2019/09/20 04:55:28.952561 [INFO] Number of way refs = 232999
2019/09/20 04:55:28.952564 [INFO] Number of reverse way refs = 20486
2019/09/20 04:55:28.952568 [INFO] Unique Node Strings (names, refs, etc.) = 29706
2019/09/20 04:55:28.952571 [INFO] Unique Strings (names, refs, etc.) = 6215216
2019/09/20 04:55:29.038573 [INFO] Creating graph edges from ways...
2019/09/20 04:56:44.690811 [INFO] Finished with 184595932 graph edges
2019/09/20 04:56:45.089228 [INFO] Sorting graph...
2019/09/20 05:15:13.774814 [INFO] Finished with 143106983 graph nodes
2019/09/20 05:15:14.412373 [INFO] Reclassifying link graph edges...
2019/09/20 05:19:25.884622 [INFO] Class: 0 exit count = 105743
2019/09/20 05:19:25.886992 [INFO] Class: 1 exit count = 126647
2019/09/20 05:19:25.887002 [INFO] Class: 2 exit count = 178604
2019/09/20 05:19:25.887006 [INFO] Class: 3 exit count = 200381
2019/09/20 05:19:25.887009 [INFO] Class: 4 exit count = 128184
2019/09/20 05:19:25.887012 [INFO] Class: 5 exit count = 22013
2019/09/20 05:19:25.887016 [INFO] Class: 6 exit count = 41969
2019/09/20 05:19:25.887019 [INFO] Class: 7 exit count = 0
2019/09/20 05:35:12.370935 [INFO] Finished with 679490 reclassified.  Turn channel count = 347457
2019/09/20 05:35:12.795370 [INFO] Reclassifying ferry connection graph edges...
2019/09/20 05:44:16.240703 [INFO] Finished ReclassifyFerryEdges: ferry_endpoint_count = 1967, 66550 edges reclassified.
2019/09/20 05:44:16.840918 [INFO] Building 46814 tiles with 4 threads...
2019/09/20 05:44:16.849241 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 05:44:16.849305 [WARN] Time zone db /home/valservice/valhalla_router/valhalla_tiles/timezones.sqlite not found.  Not saving time zone information.
2019/09/20 05:44:16.849942 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 05:44:16.849999 [WARN] Time zone db /home/valservice/valhalla_router/valhalla_tiles/timezones.sqlite not found.  Not saving time zone information.
2019/09/20 05:44:16.850008 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 05:44:16.850122 [WARN] Time zone db /home/valservice/valhalla_router/valhalla_tiles/timezones.sqlite not found.  Not saving time zone information.
2019/09/20 05:44:16.850642 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 05:44:16.850664 [WARN] Time zone db /home/valservice/valhalla_router/valhalla_tiles/timezones.sqlite not found.  Not saving time zone information.
2019/09/20 05:45:33.250480 [WARN] Speed = 150 wayId= 706566228
2019/09/20 05:45:33.253880 [WARN] Speed = 150 wayId= 706566227
2019/09/20 05:45:33.698331 [WARN] Speed = 150 wayId= 706566228
2019/09/20 05:45:34.152666 [WARN] Speed = 150 wayId= 706566227
2019/09/20 05:45:34.152740 [WARN] Speed = 150 wayId= 166291289
2019/09/20 05:49:00.679746 [WARN] Speed = 150 wayId= 25469747
2019/09/20 05:49:00.680909 [WARN] Speed = 150 wayId= 25469747
2019/09/20 05:49:00.680926 [WARN] Speed = 150 wayId= 25469744
2019/09/20 05:49:00.680939 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:00.680970 [WARN] Speed = 150 wayId= 25469744
2019/09/20 05:49:00.680984 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:00.681014 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:01.976105 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:01.976818 [WARN] Speed = 150 wayId= 625523473
2019/09/20 05:49:01.977791 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:01.977816 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:01.977926 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:01.977957 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:04.030334 [WARN] Speed = 150 wayId= 625497423
2019/09/20 05:49:04.032433 [WARN] Speed = 150 wayId= 625497422
2019/09/20 05:49:04.032468 [WARN] Speed = 150 wayId= 625523473
2019/09/20 05:49:04.032612 [WARN] Speed = 150 wayId= 625497423
2019/09/20 05:49:05.929781 [WARN] Speed = 150 wayId= 136184117
2019/09/20 05:49:05.929845 [WARN] Speed = 150 wayId= 136184114
2019/09/20 05:49:05.929881 [WARN] Speed = 150 wayId= 136184117
2019/09/20 05:49:05.929907 [WARN] Speed = 150 wayId= 451894612
2019/09/20 05:49:05.929925 [WARN] Speed = 150 wayId= 136184114
2019/09/20 05:49:06.530783 [WARN] Speed = 150 wayId= 625523474
2019/09/20 05:49:06.530862 [WARN] Speed = 150 wayId= 451894612
2019/09/20 05:49:06.895728 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:06.895839 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:06.957816 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:06.957894 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.579730 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.579794 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.580417 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.580437 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.581975 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582010 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582710 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582732 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582760 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582784 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582804 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582813 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582835 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:07.582851 [WARN] Speed = 150 wayId= 43053918
2019/09/20 05:49:10.081811 [WARN] Speed = 150 wayId= 625497422
2019/09/20 05:49:10.081909 [WARN] Speed = 150 wayId= 647627173
2019/09/20 05:49:10.144682 [WARN] Speed = 150 wayId= 647627173
2019/09/20 05:49:10.144766 [WARN] Speed = 150 wayId= 625523474
2019/09/20 05:57:51.964469 [WARN] Exceeding maximum.  Average speed: 164
2019/09/20 05:57:52.024826 [WARN] Exceeding maximum.  Average speed: 164
2019/09/20 06:16:54.002089 [WARN] Too many names for edgeindex: 46920396
2019/09/20 06:25:41.639680 [WARN] Restrictions mask exceeds allowable limit on wayid: 28653224
2019/09/20 06:25:41.641155 [WARN] Restrictions mask exceeds allowable limit: 384
2019/09/20 06:25:41.868836 [WARN] Restrictions mask exceeds allowable limit on wayid: 154988355
2019/09/20 06:25:41.868912 [WARN] Restrictions mask exceeds allowable limit: 510
2019/09/20 06:25:42.201048 [WARN] Restrictions mask exceeds allowable limit on wayid: 548393055
2019/09/20 06:25:42.201143 [WARN] Restrictions mask exceeds allowable limit: 385
2019/09/20 06:25:42.202019 [WARN] Restrictions mask exceeds allowable limit on wayid: 654568890
2019/09/20 06:25:42.202044 [WARN] Restrictions mask exceeds allowable limit: 510
2019/09/20 06:25:42.202130 [WARN] Restrictions mask exceeds allowable limit on wayid: 640540601
2019/09/20 06:25:42.202151 [WARN] Restrictions mask exceeds allowable limit: 386
2019/09/20 06:29:59.363453 [ERROR] Failed tile 2/746780/0: std::bad_alloc
2019/09/20 07:54:40.612774 [WARN] Restrictions mask exceeds allowable limit on wayid: 180572036
2019/09/20 07:54:40.614394 [WARN] Restrictions mask exceeds allowable limit: 1021
2019/09/20 07:54:40.614447 [WARN] Restrictions mask exceeds allowable limit on wayid: 389186706
2019/09/20 07:54:40.614460 [WARN] Restrictions mask exceeds allowable limit: 1022
2019/09/20 07:54:40.614586 [WARN] Restrictions mask exceeds allowable limit on wayid: 389186595
2019/09/20 07:54:40.614602 [WARN] Restrictions mask exceeds allowable limit: 1019
2019/09/20 07:54:40.614662 [WARN] Restrictions mask exceeds allowable limit on wayid: 389186596
2019/09/20 07:54:40.614676 [WARN] Restrictions mask exceeds allowable limit: 1022
2019/09/20 08:15:43.974606 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:43.993214 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:43.994460 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:44.109252 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:44.109299 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:44.287565 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:44.287635 [WARN] Speed = 150 wayId= 166291289
2019/09/20 08:15:53.969153 [INFO] Finished
2019/09/20 08:15:53.988359 [INFO] Node Count = 107436401
2019/09/20 08:15:53.988390 [INFO] Directed Edge Count = 276320329
2019/09/20 08:15:53.988394 [INFO] EdgeInfo Count = 139395830
2019/09/20 08:15:53.995992 [INFO] Enhancing local graph...
2019/09/20 08:15:54.540517 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 08:15:54.540635 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 08:15:54.540618 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 08:15:54.540724 [WARN] Admin db /home/valservice/valhalla_router/valhalla_tiles/admins.sqlite not found.  Not saving admin information.
2019/09/20 08:16:42.649539 [WARN] Exceeding maximum.  Average speed: 163
2019/09/20 08:16:42.799631 [WARN] Exceeding maximum.  Average speed: 163
2019/09/20 08:17:18.714367 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:18.789828 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:18.933168 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.135880 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.136087 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.139162 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.161092 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.575404 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.575537 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.576151 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.577083 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.619140 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.619413 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.619569 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.640548 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.640652 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.641104 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:19.641135 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:52.557588 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:17:56.802258 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:21:27.861754 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:21:36.240624 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:22:10.957563 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:22:11.389653 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:26:04.968277 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:26:26.440790 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:26:56.995036 [WARN] Exceeding maximum.  Average speed: 150
2019/09/20 08:26:57.837118 [WARN] Exceeding maximum.  Average speed: 150
2019/09/20 08:30:07.530628 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:30:07.530750 [WARN] Exceeding maximum.  Average speed: 141
2019/09/20 08:35:52.876626 [WARN] Exceeding maximum.  Average speed: 150
2019/09/20 08:35:53.085842 [WARN] Exceeding maximum.  Average speed: 150
Command terminated by signal 11

--- a/src/mjolnir/graphenhancer.cc
+++ b/src/mjolnir/graphenhancer.cc
@@ -397,8 +397,8 @@ void UpdateTurnLanes(const OSMData& osmdata,
     auto index = tilebuilder.turnlanes_offset(idx);
     std::string turnlane_tags = osmdata.name_offset_map.name(index);
     std::string str = TurnLanes::GetTurnLaneString(turnlane_tags);
-
     std::vector<uint16_t> enhanced_tls = TurnLanes::lanemasks(str);
+
     bool bUpdated = false;
     // handle [left, none, none, right] --> [left, straight, straight, right]
     // handle [straight, none, [straight, right], right] --> [straight, straight, [straight, right],
@@ -432,11 +432,9 @@ void UpdateTurnLanes(const OSMData& osmdata,
     if (!bUpdated) {
       // handle [none, none, right] --> [straight, straight, right]
       enhanced_tls = TurnLanes::lanemasks(str);
-      std::vector<uint16_t>::reverse_iterator r_it = enhanced_tls.rbegin(); // note reverse iterator
-      std::vector<uint16_t>::reverse_iterator r_it_e = enhanced_tls.rend();
-      if (((*r_it & kTurnLaneRight) || (*r_it & kTurnLaneSharpRight) ||
-           (*r_it & kTurnLaneSlightRight)) &&
-          (*r_it_e == kTurnLaneEmpty || *r_it_e == kTurnLaneNone)) {
+      if (((enhanced_tls.back() & kTurnLaneRight) || (enhanced_tls.back() & kTurnLaneSharpRight) ||
+           (enhanced_tls.back() & kTurnLaneSlightRight)) &&
+          (enhanced_tls.front() == kTurnLaneEmpty || enhanced_tls.front() == kTurnLaneNone)) {
 
         std::set<Turn::Type> outgoing_turn_type;
         GetTurnTypes(directededge, idx, outgoing_turn_type, startnodeinfo, tilebuilder, reader, lock);
@@ -461,11 +459,10 @@ void UpdateTurnLanes(const OSMData& osmdata,
     if (!bUpdated) {
       // handle [straight, straight, none] --> [straight, straight, straight]
       enhanced_tls = TurnLanes::lanemasks(str);
-      std::vector<uint16_t>::iterator it = enhanced_tls.begin();
-      std::vector<uint16_t>::iterator it_e = enhanced_tls.end();
-      if ((*it & kTurnLaneThrough) && (*it_e == kTurnLaneEmpty || *it_e == kTurnLaneNone)) {
+      if ((enhanced_tls.front() & kTurnLaneThrough) &&
+          (enhanced_tls.back() == kTurnLaneEmpty || enhanced_tls.back() == kTurnLaneNone)) {
         uint16_t previous = 0u;
-        for (; it != enhanced_tls.end(); it++) {
+        for (auto it = enhanced_tls.begin(); it != enhanced_tls.end(); it++) {
           if ((*it & kTurnLaneThrough) && (previous == 0u || (previous & kTurnLaneThrough))) {
             previous = *it;
           } else if (previous && (*it == kTurnLaneEmpty || *it == kTurnLaneNone)) {
@@ -489,14 +486,12 @@ void UpdateTurnLanes(const OSMData& osmdata,
     if (!bUpdated) {
       // handle [none, straight, straight] --> [straight, straight, straight]
       enhanced_tls = TurnLanes::lanemasks(str);
-
-      std::vector<uint16_t>::reverse_iterator r_it = enhanced_tls.rbegin(); // note reverse iterator
-      std::vector<uint16_t>::reverse_iterator r_it_e = enhanced_tls.rend();
-
       uint16_t previous = 0u;
-      if ((*r_it & kTurnLaneThrough) && (*r_it_e == kTurnLaneEmpty || *r_it_e == kTurnLaneNone)) {
-        for (; r_it != enhanced_tls.rend(); r_it++) {
-          if ((*r_it & kTurnLaneThrough) && (previous == 0u || (previous & kTurnLaneThrough))) {
+      if ((enhanced_tls.back() & kTurnLaneThrough) &&
+          (enhanced_tls.front() == kTurnLaneEmpty || enhanced_tls.front() == kTurnLaneNone)) {
+        for (auto r_it = enhanced_tls.rbegin(); r_it != enhanced_tls.rend(); r_it++) {
+          if ((enhanced_tls.back() & kTurnLaneThrough) &&
+              (previous == 0u || (previous & kTurnLaneThrough))) {
             previous = *r_it;
           } else if (previous && (*r_it == kTurnLaneEmpty || *r_it == kTurnLaneNone)) {
             *r_it = kTurnLaneThrough;

